### PR TITLE
Issue #535: Update completion's README.md for bash shell for macos

### LINKED ISSUE #535: Update completion's `README.md` for bash shell for macos - doesn't work ATM

### PR DESCRIPTION
**Pull request by amalota**
closes #535

### CODE REVIEW

**Comment by amalota** [comment]
Not sure if these changes or sth else while i was trying to get bash completion to work, but zsh completion stopped working, this PR is blocked until i fix it/figure out what's caused it.

**Comment by amalota** [comment]
**Update**
Turns out that running `brew install git bash-completion` is vital for git-machete bash completion to work but at the same time brakes git-machete zsh completion, @PawelLipski wdyt? Is that to be expected or should i keep looking for a better solution?

**Comment by PawelLipski** [comment]
> Turns out that running brew install git bash-completion is vital for git-machete bash completion to work but at the same time brakes git-machete zsh completion

Wow that's interesting lol 😯 

Okay so from practical point of view, I'd note that:
1. every Mac user I know uses (oh my) zsh,
2. literally every person I know, Mac or otherwise, uses just a single shell (bash/zsh/fish), never two at the same time

So I think we can safely ignore that problem!

**Review by PawelLipski — COMMENTED** [review]
Generally approving then!

File: completion/README.md [hunk 1]
```diff
@@ -6,6 +6,19 @@ Make sure you have bash completion installed (with `brew install bash-completion
 
 `brew install git-machete` automatically installs bash completion files for `git machete`.
 
+If the bash completion doesn't work:
+1. Install git bash completions with `brew install git bash-completion`
+2. Add the below code to `~/.bash_profile` and reload the shell
+```shell script
+[ -f /usr/local/etc/bash_completion ] && . /usr/local/etc/bash_completion || {
```
Comment (PawelLipski) [inline completion/README.md]: Actually... where is it taken from? or is it a work of yours?

File: completion/README.md [hunk 2]
```diff
@@ -6,6 +6,19 @@ Make sure you have bash completion installed (with `brew install bash-completion
...
+    # if not found in /usr/local/etc, try the brew --prefix location
+    [ -f "$(brew --prefix)/etc/bash_completion.d/git-completion.bash" ] && \
+    . $(brew --prefix)/etc/bash_completion.d/git-completion.bash
```
Comment (PawelLipski) [inline completion/README.md]: Nit: indent the continuation lines like this one by one level, so that it's clear it's a continuation

Comment (PawelLipski) [reply completion/README.md]: Or even better, extra indent + move `&&` from the previous line to the start of this line

File: completion/README.md (hunk 1 above)
Comment (amalota) [reply completion/README.md]: It's taken from here: https://gist.github.com/romansavrulin/41e55fba693b4025ed693559083bc3a0 then i deduced i need to to the same for git-machete bash completions to work

File: completion/README.md (hunk 2 above)
Comment (amalota) [reply completion/README.md]: @PawelLipski better now?

image

Comment (PawelLipski) [reply completion/README.md]: Huh but this isn't a valid shell script now! Try executing that

Comment (PawelLipski) [reply completion/README.md]: Huh but this isn't a valid shell script now! Try executing that

Comment (PawelLipski) [reply completion/README.md]: Trailing `\`s are still needed

**Review by PawelLipski — CHANGES_REQUESTED** [review]
as in comments

**Comment by amalota** [comment]
> as in comments

@PawelLipski fixed

File: completion/README.md [hunk 3]
```diff
@@ -6,6 +6,20 @@ Make sure you have bash completion installed (with `brew install bash-completion
 
 `brew install git-machete` automatically installs bash completion files for `git machete`.
 
+If the bash completion doesn't work:
+1. Install git bash completions with `brew install git bash-completion`
+2. Add the below code to `~/.bash_profile` and reload the shell
+```shell script
+[ -f /usr/local/etc/bash_completion ] && . /usr/local/etc/bash_completion \
+  || # if not found in /usr/local/etc, try the brew --prefix location
```
Comment (PawelLipski) [inline completion/README.md:14]: Does it execute cleanly now? 🤔

Comment (amalota) [reply completion/README.md:14]: Yeah, i tested it and it works OK

Comment (PawelLipski) [reply completion/README.md:14]: Okay! Indeed; I thought that `\` would be necessary after `||` as well... but it isn't

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**